client/engine/store/mocktore_test.go

```diff
@@ -0,0 +1,94 @@
+package store
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel/state"
+	directfund "github.com/statechannels/go-nitro/protocols/direct-fund"
+	"github.com/statechannels/go-nitro/types"
+)
+
+func TestNewMockStore(t *testing.T) {
+	NewMockStore([]byte{'a', 'b', 'c'})
+}
+
+func TestSetGetObjective(t *testing.T) {
+	ms := NewMockStore([]byte{})
+	got, ok := ms.GetObjectiveById("404")
```

Comment:
Looking at this, I am wondering what benefits a more strictly typed store API would be.

Accepting any old string makes it easier to get garbage input. Defining something like `type ObjectiveId = string` might be a net positive.

Reply:
created #153

Reply:
... and closed. `GetObjectiveById(id)` already expects a `protocols.Objective` typed argument. The string literal passes this type check, but string-typed variables passed in do not.